lib/phoenix/token.ex

```diff
@@ -0,0 +1,49 @@
+defmodule Phoenix.Token do
+  @moduledoc """
+  Will generate, verify, and decrypt tokens for use in channel or API 
+  authentication. Typically you will want to store the basic information
+  you need to authorize a user. Typically as simple as a user id in a 
+  database. 
+
+      iex> user_id = 1
+      iex> token = encrypt(user_id, endpoint)
+      iex> user_id == decrypt(token, endpoint)
+      true
+  """
+  alias Plug.Crypto.KeyGenerator
+  alias Plug.Crypto.MessageEncryptor
+
+  @doc """
+    Encrypts your data into a token you can send down to clients
+  """
+  def gen_token(context, data) do
+    {secret, sign_secret} = get_endpoint_for(context) |> encryptor()
+    MessageEncryptor.encrypt_and_sign(data, secret, sign_secret)
+  end
+
+  @doc """
+    Decrypts the token into the originaly present data.
+  """
+  def verify_token(context, token) do
+    {secret, sign_secret} = get_endpoint_for(context) |> encryptor()
+    case MessageEncryptor.verify_and_decrypt(token, secret, sign_secret) do
+      :error -> :error
+      {:ok, data} -> 
+    end
+  end
+
+  defp get_endpoint_for(conn = %Plug.Conn{}), do: Phoenix.Controller.endpoint_module(conn)
```

Comment:
I also personally am moving away from `_for`, even though my Ruby code is littered with it. `get_endpoint` is just fine